File: org.tec.datastructures/src/org/tec/datastructures/linear/CircularList.java
package org.tec.datastructures.linear;

import org.tec.datastructures.nodes.SimpleNode;

public class CircularList<T extends Comparable<T>>{
	
	private SimpleNode<T> head;
	
	public CircularList() {
		this(null,null);
	}
	
	public CircularList(SimpleNode<T> Head, SimpleNode<T> Next) {
		head = Head;
		if (head != null) {
			head.setNext(Next);
		}
		
	}
	
	public SimpleNode<T> getHead(){
		return this.head;
	}
	
	
	public void append (T value) {
		if (head == null){
			head = new SimpleNode<T>(value);
			head.setNext(head);
			return;
		}
		T headCopy = head.getValue();
		SimpleNode<T> headNextCopy = head.getNext();
		
		head.setValue(value);
		
		SimpleNode<T> insert = new SimpleNode<T>(headCopy);
		
		head.setNext(insert);
		insert.setNext(headNextCopy);
		
		head = insert;
	}
	
	
	public SimpleNode<T> search(T value) {
		SimpleNode<T> temp = head;
		
		if (head == null) {
			return null;
		} 
		
		do {
			if(temp.getValue().equals(value)){
				return temp;
			}
			temp = temp.getNext();
			
		}while(temp != head);
		
		return null;
	}

	public void delete(T value) {
		SimpleNode<T> temp = head;
		
		if (head == null) {
			return;
		}
		
		do {
			if(temp.getValue().equals(value)) {
				if (temp == head) {
					if (length() == 1) {
						head = null;
					}
				}
				temp.setValue(temp.getNext().getValue());
				if (temp.getNext() == head) {
					head = temp;
				}
				temp.setNext(temp.getNext().getNext());
				break;
			}
			temp = temp.getNext();
			
		}while(temp != head);
		
	}		
	
	
	public boolean isEmpty(){
		return this.head == null;
	}
	
	
	public int length() {
		SimpleNode<T> temp = head;
		if (temp == null) {
			return 0;
		}
		int result = 0;
		do {
			result++;
			temp = temp.getNext();
		}while (temp != head);
		return result;
	}

	public void clear() {
		this.head = null;
	}
	
}

File: C#/org.tec.datastructures/CircularList.cs
using System;
namespace org.tec.datastructures
{
    public class CircularList<T>
    {
        Node<T> tail;
        int lenght = 0;

        public void Append(T value)
        {
            if (tail == null)
            {
                tail = new Node<T>(value);
                tail.SetNext(tail);
                this.lenght++;
            }
            else
            {
                Node<T> nodo = tail;
                Node<T> newNode = new Node<T>(value);
                newNode.SetNext(nodo.GetNext());
                nodo.SetNext(newNode);
                tail = newNode;
                this.lenght++;

            }
        }

        public void Print()
        {
            Node<T> head = tail.GetNext();
            Node<T> temp = tail.GetNext();
            Console.WriteLine(temp.GetValue());
            temp = temp.GetNext();
            while (temp != head)
            {
                Console.WriteLine(temp.GetValue());
                temp = temp.GetNext();
            }
        }

        public void Delete(int index)
        {
            Node<T> temp = this.tail;
            int counter = 0;
            do
            {
                if (counter == index)
                {
                    temp.SetNext(temp.GetNext().GetNext());
                    this.lenght--;
                    return;
                }
                else
                {
                    temp = temp.GetNext();
                    counter++;
                }
            } while (temp.GetNext() != tail.GetNext());
            Console.WriteLine("No se pudo encontrar el elemento");
        }

        public T Get(int index)
        {
            Node<T> temp = this.tail.GetNext();
            int counter = 0;
            do
            {
                if (counter == index)
                {
                    return temp.GetValue();
                }
                else
                {
                    temp = temp.GetNext();
                    counter++;
                }
            } while (temp != this.tail.GetNext());
            throw new ArgumentOutOfRangeException();
        }

        public Boolean IsEmpty(){
            return this.tail == null;
        }

        public int Lenght(){
            return this.lenght;
        }


        public class Node<T>
        {
            Node<T> next = null;
            T value;

            public Node(T value)
            {
                this.value = value;
            }

            public Node<T> GetNext()
            {
                return next;
            }

            public void SetNext(Node<T> node)
            {
                next = node;
            }

            public T GetValue()
            {
                return this.value;
            }
        }
    }
}